### PR TITLE
Add help for doc, license and proto targets

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does

[Add help for doc targets](https://github.com/grafana/mimir/commit/27dc3a4d33819d728c6caa15a583f703977c3218)

[Add help for license targets](https://github.com/grafana/mimir/commit/7c6caf521c1c417b3ae1ad2472cb2cc513b06b51)

[Add help for proto targets](https://github.com/grafana/mimir/commit/9bf2481d93924877779a85bd26264a0ec2b96626)

#### Which issue(s) this PR fixes or relates to

Part of https://github.com/grafana/mimir/issues/2499

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -432,16 +435,18 @@ clean:
...
 
 # List all images building make targets.
 list-image-targets:
 	@echo $(UPTODATE_FILES) | tr " " "\n"
 
-clean-doc:
+clean-doc: ## Clean the config file documentation.
 	rm -f $(DOC_TEMPLATES:.template=.md)
 
+check-doc: ## Check the config file is up to date.
```
Comment (pracucci) [inline Makefile:448]:
```suggestion
check-doc: ## Check the documentation files are up to date.
```

File: Makefile [hunk 2]
```diff
@@ -432,16 +435,18 @@ clean:
...
+	@./tools/check-doc.sh
```
Comment (pracucci) [inline Makefile:449]: What's this?

File: Makefile [hunk 3]
```diff
@@ -432,16 +435,18 @@ clean:
...
-clean-protos:
+clean-protos: ## Clean protobuf files.
 	rm -rf $(PROTO_GOS)
 
 # List all images building make targets.
 list-image-targets:
 	@echo $(UPTODATE_FILES) | tr " " "\n"
 
-clean-doc:
+clean-doc: ## Clean the config file documentation.
```
Comment (pracucci) [inline Makefile:445]:
```suggestion
clean-doc: ## Clean the documentation files generated from templates.
```

**Review by pracucci — COMMENTED** [review]
Thanks! Mostly LGTM, but I don't understand the new call to `check-doc.sh`.

File: Makefile (hunk 2 above)
Comment (hi-rustin) [reply Makefile]: I submitted the wrong code, probably related to a mistake in my rebase process. Sorry for confusing.

**Comment by hi-rustin** [comment]
> Thanks! Mostly LGTM, but I don't understand the new call to `check-doc.sh`.

Thanks for your review. I submitted the wrong code, probably related to a mistake in my rebase process. Sorry for the confusing and typos.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**